### ISSUE TITLE
Incomplete/wrong breakdown?

### ISSUE THREAD

**Issue by sajith** [atlanticwave-sdx/pce#112]
Given a merged "super topology" that has three domains (amlight, sax, zaoxi) in sdx-controller, and a connection request, the connection solution should be broken down to three domains.  But instead we end up with something like this:

```json
{
    "urn:ogf:network:sdx": {
        "ingress_port": {
            "id": "urn:ogf:network:sdx:port:sax:B1:2",
            "name": "Unknown",
            "short_name": null,
            "node": "urn:ogf:network:sdx:node:sax:B1",
            "label_range": null,
            "status": null,
            "state": null,
            "private_attributes": null
        },
        "egress_port": {
            "id": "urn:ogf:network:sdx:port:sax:B1:3",
            "name": "Novi01:3",
            "node": "urn:ogf:network:sdx:node:sax:B1",
            "short_name": "B1:3",
            "label_range": [
                "100-200",
                "10001"
            ],
            "status": "up"
        }
    }
}
```

Seems that the method that finds a domain of a node is wrong.

**Comment by YufengXin** [comment]
Well, if the two ports of the connection request belong to the same domain, the breakdown will only involve the one domain.

**Comment by sajith** [comment]
Well, I _think_ that is the case, but also want to be certain. :-)

Let us have this resolved with some more tests.  I will document the results here, and probably follow up with some more unit tests.

**Comment by sajith** [comment]
Trying to summarize my understanding of the problem again: when testing sdx-controller, @congwang09 and I found that the breakdown does not look quite right.  We should have a map of per-domain breakdowns, but we do not have it.

My guess is that `TopologyManager.add_topology()` and `TopologyManager.update_topology()` do not preserve the ownership information about a node.  We need to find what topology (strings like `"urn:ogf:network:sdx:topology:amlight.net"`) a node belongs to, but once a topology has been added to TopologyManager, updated, TopologyManager no longer has that information.  

Consequently, `TopologyManager.get_domain_name()` returns `"urn:ogf:network:sdx"` (which was set by `TopologyManager.generate_id()` when the topology was initially added.

**Comment by sajith** [comment]
> My guess is that `TopologyManager.add_topology()` and `TopologyManager.update_topology()` do not preserve the ownership information about a node. We need to find what topology (strings like `"urn:ogf:network:sdx:topology:amlight.net"`) a node belongs to, but once a topology has been added to TopologyManager, updated, TopologyManager no longer has that information.

Perhaps I am wrong here.  `TopologyManager.topology_list` should have what we need, but `TopologyManager.get_domain_name()` isn't returning the right thing.

**Comment by YufengXin** [comment]
The key in  te.topology_list is topology.id, so make sure it's correct. te.get_domain_name(node) only checks on if it contains the particular node and returns topology.id if so.

**Comment by sajith** [comment]
In the tests I did (https://github.com/atlanticwave-sdx/pce/pull/113) with the three topologies we use, `get_domain_name()` returns the correct topology ID. So I guess it is possible that the wrong topology data is somehow passed around somewhere in sdx-controller.

I think stronger validation wherever possible (eventually - perhaps using [pydantic](https://docs.pydantic.dev/latest/) or similar) would be a good idea.

**Comment by sajith** [comment]
Closed in #113.
